examples/cis2-multi/src/lib.rs

```diff
@@ -1,38 +1,183 @@
 //! function for transferring an amount of a specific token type from one
 //! address to another address. An address can enable and disable one or more
-//! addresses as operators. An operator of some address is allowed to transfer
-//! any tokens owned by this address.
+//! addresses as operators with the `updateOperator` function. An operator of
+//! an address is allowed to transfer any tokens owned by this address.
+//! As follows from the CIS3 specification, the contract has a `permit`
+//! function. It is the sponsored counterpart to the `transfer/updateOperator`
+//! function and can be executed by anyone on behalf of an account given a
+//! signed message.
 //!
 //! This contract also contains an example of a function to be called when
 //! receiving tokens. In which case the contract will forward the tokens to
 //! the contract owner.
 //! This function is not very useful and is only there to showcase a simple
 //! implementation of a token receive hook.
+//!
+//! Concordium supports natively multi-sig accounts. Each account address on
+//! Concordium is controlled by one or several credential(s) (real-world
+//! identities) and each credential has one or several public-private key
+//! pair(s). Both the key and signature structures of an account are represented
+//! as two-level maps (e.g. BTreeMap<CredentialIndexU8,
+//! BTreeMap<PublicKeyIndexU8, PublicKey>> or BTreeMap<CredentialIndexU8,
+//! BTreeMap<SignatureIndexU8, Signature>>). The outer map of
+//! `BTreeMap<CredentialIndexU8, BTreeMap<PublicKeyIndexU8, PublicKey>>` has an
+//! `AccountThreshold` (number of credentials needed to sign the transaction
+//! initiated by that account) and the inner map has a `SignatureThreshold`
+//! (number of Signatures needed for a specific credential so that this
+//! credential is considered to have signed the transaction initiated by that
+//! account). The CIS3 standard supports multi-sig accounts. For basic accounts
+//! (that have exactly one credential and exactly one public key for that
+//! credential), the signaturesMaps/publicKeyMaps in this contract, will have
+//! only one value at key 0 in the inner and outer maps.
 
 #![cfg_attr(not(feature = "std"), no_std)]
 use concordium_cis2::*;
-use concordium_std::*;
+use concordium_std::{collections::BTreeMap, EntrypointName, *};
 
 /// List of supported standards by this contract address.
 const SUPPORTS_STANDARDS: [StandardIdentifier<'static>; 2] =
     [CIS0_STANDARD_IDENTIFIER, CIS2_STANDARD_IDENTIFIER];
 
+/// List of supported entrypoints by the `permit` function (CIS3 standard).
+const SUPPORTS_PERMIT_ENTRYPOINTS: [EntrypointName; 2] =
+    [EntrypointName::new_unchecked("updateOperator"), EntrypointName::new_unchecked("transfer")];
+
+/// Tag for the CIS3 Nonce event.
+pub const NONCE_EVENT_TAG: u8 = u8::MAX - 5;
+
+/// The amount of tokens to airdrop when the mint function is invoked.
+pub const MINT_AIRDROP: TokenAmountU64 = TokenAmountU64(100);
```

Comment:
Since the value is never updated in the smart contract logic. I feel it is more of a constant and does not have to be stored in the smart contract state (meant for values that are updated over the life of a smart contract).

Reply:
True, people could just initialize a new instance instead of having to compile/deploy a new module in case they decide to change the `MINT_AIRDROP` value. Ok, I will change it to use an input parameter here.

Passing the value as an input parameter is likely a bit more expensive (more logic and bigger module size) than having it hardcoded, isn't it?

Reply:
Yes, but I expect this to be hard to observe.